packages/mui-system/src/borders.js

```diff
@@ -68,7 +68,7 @@ export const borderLeftColor = style({
 });
 
 export const borderRadius = (props) => {
-  if (props.borderRadius) {
+  if (props.borderRadius !== undefined && props.borderRadius !== null) {
```

Comment:
Oh man 😄 Thanks! I will check if there are similar occurrences of this condition.

Reply:
I stopped mentioning these types of defects since people just don't want to listen. We lint against explicit implicit type coercion e.g. `if (props.borderRadius != false)` but developers repeatedly use implicit implicit type coercion e.g `if (props.borderRadius)`.

These types of bugs will continue to occur unless we recognize that implicit type coercion is a problem that needs to be addressed.